pkg/utils/common.go

```diff
@@ -67,55 +72,148 @@ func GetUkiCmdline() []string {
+
+	gzw := gzip.NewWriter(mw)
+	defer gzw.Close()
+
+	tw := tar.NewWriter(gzw)
+	defer tw.Close()
 
-	err := filepath.Walk(root, func(fp string, info os.FileInfo, err error) error {
+	// walk path
+	return filepath.Walk(src, func(file string, fi os.FileInfo, err error) error {
+
+		// return on any error
 		if err != nil {
+			return err
+		}
+
+		// return on non-regular files (thanks to [kumo](https://medium.com/@komuw/just-like-you-did-fbdd7df829d3) for this suggested update)
+		if !fi.Mode().IsRegular() {
 			return nil
 		}
-		rel, err := filepath.Rel(root, fp)
+
+		// create a new dir/file header
+		header, err := tar.FileInfoHeader(fi, fi.Name())
 		if err != nil {
-			return fmt.Errorf("failed to calculate relative path: %w", err)
+			return err
 		}
 
-		hdr := &tar.Header{
-			Name: path.Join("/", filepath.ToSlash(rel)),
-			Mode: int64(info.Mode()),
+		// update the name to correctly reflect the desired destination when untaring
+		header.Name = strings.TrimPrefix(strings.Replace(file, src, "", -1), string(filepath.Separator))
+
+		// write the header
+		if err := tw.WriteHeader(header); err != nil {
+			return err
 		}
 
-		if !info.IsDir() {
-			hdr.Size = info.Size()
+		// open files for taring
+		f, err := os.Open(file)
+		if err != nil {
+			return err
 		}
 
-		if info.Mode().IsDir() {
-			hdr.Typeflag = tar.TypeDir
-		} else {
-			hdr.Typeflag = tar.TypeReg
+		// copy file data into tar writer
+		if _, err := io.Copy(tw, f); err != nil {
+			return err
 		}
 
-		if err := tw.WriteHeader(hdr); err != nil {
-			return fmt.Errorf("failed to write tar header: %w", err)
+		// manually close here after each file operation; defering would cause each file close
+		// to wait until all operations have completed.
+		f.Close()
+
+		return nil
+	})
+}
+
+// CreateTar a imagetarball from a standard tarball
+func CreateTar(log v1.Logger, srctar, dstimageTar, imagename, architecture, OS string) error {
+
+	dstFile, err := os.Create(dstimageTar)
+	if err != nil {
+		return fmt.Errorf("Cannot create %s: %s", dstimageTar, err)
+	}
+	defer dstFile.Close()
+
+	newRef, img, err := imageFromTar(imagename, architecture, OS, func() (io.ReadCloser, error) {
+		f, err := os.Open(srctar)
+		if err != nil {
+			return nil, fmt.Errorf("cannot open %s: %s", srctar, err)
 		}
-		if !info.IsDir() {
-			f, err := os.Open(fp)
-			if err != nil {
-				return err
-			}
-			if _, err := io.Copy(tw, f); err != nil {
-				return fmt.Errorf("failed to read file into the tar: %w", err)
-			}
-			f.Close()
+		decompressed, err := containerdCompression.DecompressStream(f)
+		if err != nil {
+			return nil, fmt.Errorf("cannot open %s: %s", srctar, err)
 		}
-		return nil
+
+		return decompressed, nil
 	})
 	if err != nil {
-		return nil, fmt.Errorf("failed to scan files: %w", err)
+		return err
 	}
-	if err := tw.Close(); err != nil {
-		return nil, fmt.Errorf("failed to finish tar: %w", err)
+
+	// Lets try to load it into the docker daemon?
```

Comment:
I'd even drop it TBH - if we produce a tar it's easier for anyone to just adhere on a single type of output. Maybe what we can do is stream it to stdout so can be piped:

``
enki .... | docker load && docker push
```

Reply:
done, left the code in there in case we want to use it down the line maybe with a flag or something